lib/NodeFactory/DefaultNodeFactory.php

```diff
@@ -145,13 +145,10 @@ public function createListNode() : ListNode
         return $listNode;
     }
 
-    /**
-     * @param string[] $parts
-     */
-    public function createTableNode(array $parts, string $type, LineChecker $lineChecker) : TableNode
+    public function createTableNode(Parser\TableSeparatorLineConfig $separatorLineConfig, string $type, LineChecker $lineChecker) : TableNode
```

Comment:
Changed this crazy `$parts` array (which was kind of a bunch of indexes identifying when each column started and stoped) into a new class.